### ISSUE TITLE
test_edmPickEvents fails

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#36527]
The `test_edmPickEvents` unit test has been failing since CMSSW_12_3_X_2021-12-14-2300 with
```
===== Test "test_edmPickEvents" ====
+ edmPickEvents.py /Cosmics/Run2011A-v1/RAW 160960:277:10001082,160960:277:10001058,160960:277:10001650
+ chmod +x run_edmCopyPickMerge.sh
+ ./run_edmCopyPickMerge.sh
./run_edmCopyPickMerge.sh: line 1: DAS: command not found
----- Begin Fatal Exception 16-Dec-2021 05:30:37 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
Exception Message:
InputFileCatalog::InputFileCatalog()
An empty string specified in the fileNames parameter for input source.
----- End Fatal Exception -------------------------------------------------

---> test test_edmPickEvents had ERRORS
```

The appearance of the failure coincides with new `dasgoclient` version from https://github.com/cms-sw/cmsdist/pull/7509

**Comment by makortel** [comment]
assign analysis

**Comment by makortel** [comment]
Running the test locally in CMSSW_12_3_X_2021-12-15-1100 succeeds. I'm very confused of this error. The script `run_edmCopyPickMerge` has no `DAS` that the error complains.

**Comment by makortel** [comment]
@smuzaffar Or could there be some trickery with https://github.com/cms-sw/cmssw/blob/master/Utilities/General/ibeos/das_client in play?

**Comment by smuzaffar** [comment]
I think https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/Utilities/scripts/edmPickEvents.py (which is run by this unit test) fails and printout some error message like https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/Utilities/scripts/edmPickEvents.py#L106 . The script execute the outout of `edmPickEvents.py` and then complain about unknown `DAS` command. May be there is issue with das client itself or the services it talk to

**Comment by makortel** [comment]
I managed to accidentally reproduce locally. The `run_edmCopyPickMerge.sh` got
```
DAS error: b'2021/12/16 21:39:09 ERROR, invalid URL /dbs/prod/global/DBSReader/blocks?dataset=/Cosmics/Run2011A-v1/RAW\n'
```

**Comment by smuzaffar** [comment]
@vkuznet , we see that now dasgoclient ( run here https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/Utilities/scripts/edmPickEvents.py#L121-L136 ) now randomly fails with the message like https://github.com/cms-sw/cmssw/issues/36527#issuecomment-996209145 . Could this be related to the change https://github.com/cms-sw/cmsdist/pull/7509 ?

**Comment by makortel** [comment]
The exact command being run inwide `edmPickEvents` appears to be 
```
dasgoclient -query "file dataset=/Cosmics/Run2011A-v1/RAW run=160960 lumi=277" -json
```
(and runnig that gives me a similar error)

**Comment by vkuznet** [comment]
I need some time to understand the issue here and will do it tomorrow, but it is related with recent code refactoring which I think revealed hidden issue. Meanwhile, if you'll setup proper DBS_URL environment the issue will be gone:
```
export DBS_URL=https://cmsweb.cern.ch:8443/dbs/prod/global/DBSReader
dasgoclient -query "file dataset=/Cosmics/Run2011A-v1/RAW run=160960 lumi=277" -json
[
{"das":{"expire":1639692273,"instance":"prod/global","primary_key":"file.name","record":1,"services":["dbs3:file4dataset_run_lumi"]},"file":[{"name":"/store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root"}],"qhash":"e3520c0a67318a03b339b02affe9a3e2"}
]
```

**Comment by smuzaffar** [comment]
ok , I have updated [cms-bot](https://github.com/cms-sw/cms-bot/commit/98b3bc70cef9a6ae3767c47a8499135c54f2e029)  to set `export DBS_URL=https://cmsweb.cern.ch:8443/dbs/prod/global/DBSReader`

**Comment by vkuznet** [comment]
I provided the fix in https://github.com/cms-sw/cmsdist/pull/7522, but on a safe side you should explicitly use DBS_URL (and in this regard RUCIO_URL) before using dasgoclient. This explicit settings will ensure that DAS will use them in HTTP calls.

**Comment by makortel** [comment]
@smuzaffar What about developers/users, where would they (we) get the correct `DBS_URL` (and `RUCIO_URL`)?

**Comment by vkuznet** [comment]
@makortel , all URLs are defined in dasmaps and if these env variables are NOT set DAS will use URLs from its map. Users should not worry about it and should not bother to use explicitly these env variables. I pointed about them here to ensure that production tools may explicitly define them to force DAS tools to use them.

The URLs may change, and it happened already several times in last year, e.g. DMWM team decided to separate their production workflows from cmsweb.cern.ch, as such new URL is used cmsweb-k8s-prod for DBS/ReqMgr2, etc. The Rucio team uses its own URLcms-rucio.cern.ch, etc. On testbed we use yet another url, cmsweb-testbed. There are also test cluster for dev teams like cmsweb-test1, etc.

By default all URLs are extracted from dasmaps which I maintain, and dasgoclient fetches them periodically from github and store in user's $HOME/.dasmaps area (which again can be overwritten), etc. 

The bottom line, please do not complicated things and by default everything should work. But if you do need to test specific server, you may setup these env variable and DAS tools will follow.

**Comment by makortel** [comment]
Thanks @vkuznet for the clarification!

**Comment by smuzaffar** [comment]
closing this issue as this has been fixed in new dasgoclient . We now have a new one https://github.com/cms-sw/cmssw/issues/37052
